src/tools/perf/lib/libperf.c

```diff
@@ -1173,19 +1185,29 @@ static void ucp_perf_test_destroy_eps(ucx_perf_context_t* perf)
                                   UCP_EP_CLOSE_MODE_FLUSH);
 
             if (UCS_PTR_IS_PTR(req)) {
-                do {
-                    ucp_worker_progress(perf->ucp.tctx[i].perf.ucp.worker);
-                    status = ucp_request_check_status(req);
-                } while (status == UCS_INPROGRESS);
-
-                ucp_request_release(req);
+                reqs[num_in_prog++] = req;
             } else if (UCS_PTR_STATUS(req) != UCS_OK) {
                 ucs_warn("failed to close ep %p on thread %d: %s\n",
                          perf->ucp.tctx[i].perf.ucp.ep, i,
                          ucs_status_string(UCS_PTR_STATUS(req)));
             }
         }
     }
+
+    while (num_in_prog != 0) {
+        ucp_perf_worker_progress(perf);
+        for (i = 0; i < thread_count; ++i) {
+            if (!UCS_PTR_IS_PTR(reqs[i])) {
```

Comment:
loop until num_in_prog (instead of thread_count)
then can remove the "if" here

Reply:
fixed